modules/cfe_testcase/src/fs_util_test.c

```diff
@@ -0,0 +1,161 @@
+                      CFE_FS_INVALID_PATH);
+    UtAssert_INT32_EQ(CFE_FS_ParseInputFileName(NameBuf, Name, 0, CFE_FS_FileCategory_SCRIPT), CFE_FS_BAD_ARGUMENT);
+
+    UtAssert_INT32_EQ(
+        CFE_FS_ParseInputFileNameEx(OutNameBuf, InNameBuf, sizeof(OutNameBuf), sizeof(InNameBuf), Name, Path, Ext),
+        CFE_SUCCESS);
+    UtAssert_StrCmp(ExpectedBuf, OutNameBuf, "Parse Input EX: %s", OutNameBuf);
+    UtAssert_INT32_EQ(
+        CFE_FS_ParseInputFileNameEx(OutNameBuf, NULL, sizeof(OutNameBuf), sizeof(InNameBuf), Name, Path, Ext),
+        CFE_SUCCESS);
+    UtAssert_StrCmp(ExpectedName, OutNameBuf, "Parse Input EX: %s", OutNameBuf);
+    UtAssert_INT32_EQ(CFE_FS_ParseInputFileNameEx(OutNameBuf, InNameBuf, sizeof(OutNameBuf), 0, Name, Path, Ext),
+                      CFE_SUCCESS);
+    UtAssert_StrCmp(ExpectedName, OutNameBuf, "Parse Input EX: %s", OutNameBuf);
+    UtAssert_INT32_EQ(
+        CFE_FS_ParseInputFileNameEx(OutNameBuf, InNameBuf, sizeof(OutNameBuf), sizeof(InNameBuf), NULL, Path, Ext),
+        CFE_SUCCESS);
+    UtAssert_StrCmp(ExpectedBuf, OutNameBuf, "Parse Input EX: %s", OutNameBuf);
+
+    UtAssert_INT32_EQ(
+        CFE_FS_ParseInputFileNameEx(NULL, InNameBuf, sizeof(OutNameBuf), sizeof(InNameBuf), Name, Path, Ext),
+        CFE_FS_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(CFE_FS_ParseInputFileNameEx(OutNameBuf, InNameBuf, 0, sizeof(InNameBuf), Name, Path, Ext),
+                      CFE_FS_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(
+        CFE_FS_ParseInputFileNameEx(OutNameBuf, NULL, sizeof(OutNameBuf), sizeof(InNameBuf), NULL, Path, Ext),
+        CFE_FS_INVALID_PATH);
+    UtAssert_INT32_EQ(CFE_FS_ParseInputFileNameEx(OutNameBuf, InNameBuf, sizeof(OutNameBuf), 0, NULL, Path, Ext),
+                      CFE_FS_INVALID_PATH);
+}
+
+void TestFileName(void)
+{
+    const char Path[] = "/func/FileName.test";
+    char       Name[OS_MAX_FILE_NAME];
+    const char ExpectedName[] = "FileName.test";
+
+    UtPrintf("Testing: CFE_FS_ExtractFilenameFromPath");
+
+    UtAssert_INT32_EQ(CFE_FS_ExtractFilenameFromPath(Path, Name), CFE_SUCCESS);
+    UtAssert_StrCmp(Name, ExpectedName, "Extract Filename: %s", Name);
+
+    UtAssert_INT32_EQ(CFE_FS_ExtractFilenameFromPath(NULL, Name), CFE_FS_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(CFE_FS_ExtractFilenameFromPath(Path, NULL), CFE_FS_BAD_ARGUMENT);
+}
+
+/* FT helper stub compatible with background file write DataGetter */
+bool FS_DataGetter(void *Meta, uint32 RecordNum, void **Buffer, size_t *BufSize)
+{
+    *Buffer  = NULL;
+    *BufSize = 0;
+    return true;
+}
+
+/* FT helper stub compatible with background file write OnEvent */
+void FS_OnEvent(void *Meta, CFE_FS_FileWriteEvent_t Event, int32 Status, uint32 RecordNum, size_t BlockSize,
+                size_t Position)
+{
+}
+
+CFE_FS_FileWriteMetaData_t State;
+
+void TestFileDump(void)
+{
+    CFE_FS_FileWriteMetaData_t State;
```

Comment:
Since the documentation specifically says this object is not supposed to be on the stack, should we at least add a "static" keyword?

It's OK because of the wait, but this technically still does not comply with the API documentation this way.